### PR TITLE
[ACCOUNT-2517] feat: shop healthcheck

### PR DESCRIPTION
**Pull request by hschoenenberger**

### CODE REVIEW

File: controllers/front/apiV2ShopHealthCheck.php [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+use PrestaShop\Module\PsAccounts\Account\Session\ShopSession;
+use PrestaShop\Module\PsAccounts\Account\Token\NullToken;
+use PrestaShop\Module\PsAccounts\Account\Token\Token;
+use PrestaShop\Module\PsAccounts\Api\Client\AccountsClient;
+use PrestaShop\Module\PsAccounts\Api\Controller\AbstractShopRestController;
+use PrestaShop\Module\PsAccounts\Provider\OAuth2\Oauth2Client;
+use PrestaShop\Module\PsAccounts\Provider\OAuth2\ShopProvider;
+use PrestaShop\Module\PsAccounts\Service\PsAccountsService;
+
+class ps_AccountsApiV2ShopHealthCheckModuleFrontController extends AbstractShopRestController
```
Comment (hschoenenberger) [inline controllers/front/apiV2ShopHealthCheck.php:32]: Api versioning to be discussed here

File: controllers/front/apiV2ShopHealthCheck.php [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+     * @param ShopHealthCheckRequest $request
+     *
+     * @return array
+     *
+     * @throws Exception
+     */
+    public function show(Shop $shop, ShopHealthCheckRequest $request)
+    {
+        // refreshing one of firebase tokens will trigger a global refresh
+        $firebaseShopToken = $request->autoheal ?
```
Comment (atourneriePresta) [inline controllers/front/apiV2ShopHealthCheck.php:105]: Why don't we just refresh token when it expires ? 
`GetOrRefreshToken`
The is no use of getting old token not refreshed to debug no?

File: controllers/front/apiV2ShopHealthCheck.php [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+//            'psVersion' => _PS_VERSION_,
+//            'phpVersion' => phpversion(),
+//        ];
+
+        return [
+            'oauth2Client' => $this->oauth2Client->exists(),
+            'shopLinked' => (bool) $this->linkShop->getShopUuid(),
+            'isSsoEnabled' => $this->psAccountsService->getLoginActivated(),
+            'oauthToken' => $this->tokenInfos($shopToken),
+            'firebaseUserToken' => $this->tokenInfos($firebaseOwnerToken),
```
Comment (atourneriePresta) [inline controllers/front/apiV2ShopHealthCheck.php:124]: It's named firebaseOwnerToken

**Review by atourneriePresta — CHANGES_REQUESTED** [review]

File: controllers/front/apiV2ShopHealthCheck.php (hunk 2 above)
Comment (hschoenenberger) [reply controllers/front/apiV2ShopHealthCheck.php:105]: I suggest you add a comment on the EventModeling board

File: controllers/front/apiV2ShopHealthCheck.php (hunk 3 above)
Comment (hschoenenberger) [inline controllers/front/apiV2ShopHealthCheck.php:124]:
```suggestion
            'firebaseOwnerToken' => $this->tokenInfos($firebaseOwnerToken),
```

**Review by atourneriePresta — APPROVED** [review]

**Pull request merged**